### PR TITLE
RIA-7439 ho_change_direction_due_date

### PR DESCRIPTION
**Pull request by ikirsanov**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RIA-7439

### Change description ###
Document generation for Internal HO Direction due date changed: ADA and non-ADA scenarios
Unit and functional tests

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/InternalHoChangeDirectionDueDateLetterGenerator.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                asylumCase,
+                internalChangeDirectionDueDateLetter,
+                NOTIFICATION_ATTACHMENT_DOCUMENTS,
+                DocumentTag.INTERNAL_HO_CHANGE_DIRECTION_DUE_DATE_LETTER
+        );
+
+        return new PreSubmitCallbackResponse<>(asylumCase);
+    }
+
+    private boolean isDirectionPartyRespondent(AsylumCase asylumCase) {
```
Comment (hussain-faruk) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/InternalHoChangeDirectionDueDateLetterGenerator.java:77]: Maybe we can move this to asylumCaseUtils for future use and avoid code dupes

**Review by hussain-faruk — APPROVED** [review]

**Review by hussain-faruk — CHANGES_REQUESTED** [review]
tests needed for added method in aslym case utils

**Review by hussain-faruk — DISMISSED** [review]

**Review by hussain-faruk — APPROVED** [review]

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by jamieclarkeHMCTS — APPROVED** [review]

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by jamieclarkeHMCTS — APPROVED** [review]

**Pull request merged**